app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -241,6 +242,110 @@ public GetMoodTask(String filterBy){
         }
     }
 
+    public static class AddUserTask extends AsyncTask<User, Void, Void> { //TODO: What if username already exists within the database? We need usernames to be UNIQUE for proper sorting.
+
+        @Override
+        protected Void doInBackground(User ... users ) {
+            if(android.os.Debug.isDebuggerConnected())
+                android.os.Debug.waitForDebugger();
+            verifySettings();
+
+            for (User user : users) {
+                Index index = new Index.Builder(user).index(groupIndex).type(typeUser).build();
+
+                try {
+                    // where is the client?
+                    DocumentResult result = client.execute(index);
+                    if (result.isSucceeded()){
+                        user.setId(result.getId());
+                    }
+                    else{
+                        Log.i("Error", "Elasticsearch was not able to create user");
+                    }
+                }
+                catch (Exception e) {
+                    Log.i("Error", "The application was not able to create the user");
+                }
+            }
+            return null;
+        }
+    }
+
+    public static class DeleteUserTask extends AsyncTask<User, Void, Void> {
+
+        @Override
+        protected Void doInBackground(User ... users ) {
```

Comment:
Why is this also called doInBackground? What are we doing in background? It has the same name as the other one, why do they not conflict?